client/transport/src/ws/mod.rs

```diff
@@ -404,6 +400,67 @@ impl WsTransportClientBuilder {
 		}
 		err.unwrap_or(Err(WsHandshakeError::NoAddressFound(target.host)))
 	}
+
+	/// Try to establish the connection over the given data stream.
+	pub async fn build_with_stream<T>(
```

Comment:
This should be added to the `WsClientBuilderBuilder` for a uniform API I reckon.

Reply:
ah my bad it was added on that type